src/workers/python/papyros/papyros.py

```diff
@@ -112,41 +98,25 @@ def pre_run(self, source_code, mode="exec", top_level_await=False):
         self.override_globals()
         return super().pre_run(source_code, mode=mode, top_level_await=top_level_await)
 
-    def execute(self, code_obj, source_code, mode=None):  # noqa
-        return eval(code_obj, self.console.locals)  # noqa
-
-    async def run_async(self, source_code, mode="exec", top_level_await=True, filename="my_code.py"):
-        """
-        Mostly a copy of the parent `run_async` with some key differences
-        We use `await ft` in case of an exception, because `serialize_traceback` isn't async.
-        We call pre_run within to handle its exceptions within this try-block too
-        As it will rethrow the SyntaxError (@see serialize_syntax_error)
-        """
-        self.set_file(filename, source_code)
-        try:
-            with self._execute_context(source_code):
-                try:
-                    await self.install_imports(source_code, ignore_missing=False)
-                    code_obj = self.pre_run(
-                        source_code, mode, top_level_await=top_level_await)
-                    if code_obj:
-                        result = self.execute(code_obj, source_code, mode)
-                        while isinstance(result, Awaitable):
-                            result = await result
-                        return result
-                except:
-                    # Let `_execute_context` and `serialize_traceback`
-                    # handle the exception
-                    raise
-        except KeyboardInterrupt:
-            self.callback("interrupt", data="KeyboardInterrupt", contentType="text/plain")
-
-    def serialize_syntax_error(self, exc, source_code):
+    async def run_async(self, source_code, mode="exec", top_level_await=True):
+        with self._execute_context():
+            try:
+                await self.install_imports(source_code, ignore_missing=False)
```

Comment:
Since the result from `execute` is already awaited, maybe you could override it with an async method and put this in there to avoid copying run_async?

Reply:
Is overriding a non-async method with an async one considered good practica? Or do you expect me to return a coroutine object instead of changing the method definition?

Reply:
I'm guessing it'd be OK to override with async.